integration/client/imagerules/imageroleauthorizations_test.go

```diff
@@ -0,0 +1,358 @@
+	for _, ni := range details.NestedImages {
+		ira.ImageSelector.NamePatterns = append(ira.ImageSelector.NamePatterns, ni.ImageName)
+		ira.ImageSelector.NamePatterns = append(ira.ImageSelector.NamePatterns, ni.Digest)
+	}
+	err = kclient.Update(ctx, ira)
+	require.NoError(t, err, "should not error while updating IRA")
+
+	// Ensure that the selector matches the image now
+	err = imageselector.MatchImage(ctx, kclient, c.GetNamespace(), tagName, id, image.Digest, ira.ImageSelector)
+	require.NoError(t, err, "should not error while matching image")
+
+	details, err = c.ImageDetails(ctx, id, &client.ImageDetailsOptions{IncludeNested: true})
+	require.NoError(t, err, "should not error while getting image details")
+
+	ra, err := imagerules.GetAuthorizedPermissions(ctx, kclient, c.GetNamespace(), tagName, image.Digest)
+	require.NoError(t, err, "should not error while getting authorized permissions")
+
+	// replace name in granted permissions
+	for i := range ra {
+		ra[i].ServiceName = "rootapp"
+	}
+
+	missing, granted := internalv1.GrantsAll(c.GetNamespace(), details.Permissions, ra)
+	require.True(t, granted, "should have granted permissions, but have missing: %#v\nr: %#v\ng:%#v", missing, details.Permissions, ra)
+
+	app = createWaitLoop(ctx, blueprint.DeepCopy())
+	require.Equal(t, 0, len(app.Status.Staged.ImagePermissionsDenied), "should have 1 denied permissions: %#v", app.Status.Staged.ImagePermissionsDenied)
+
+	app = helper.WaitForObject(t, helper.Watcher(t, c), &apiv1.AppList{}, app, func(obj *apiv1.App) bool {
+		return obj.Status.AppStatus.Containers["rootapp"].Ready
+	})
+
+	// When parent app is ready, the nested app should not, because it still has denied permissions
+	nestedApp, err := c.AppGet(ctx, app.Status.AppStatus.Acorns["foo"].AcornName)
+	require.NoError(t, err, "should not error while getting nested app")
+
+	nestedApp = helper.WaitForObject(t, helper.Watcher(t, c), &apiv1.AppList{}, nestedApp, func(obj *apiv1.App) bool {
+		return obj.Status.Staged.PermissionsChecked
+	})
+
+	require.Equal(t, 1, len(nestedApp.Status.Staged.ImagePermissionsDenied), "should have 1 denied permission (foo.awsapp)")
+	expectedDeniedPermissionsFooAwsapp := []internalv1.Permissions{
+		{
+			ServiceName: nestedImageTagName,
+			Rules: []internalv1.PolicyRule{
+				{
+					PolicyRule: rbacv1.PolicyRule{
+						APIGroups: []string{"aws.acorn.io"},
+						Verbs:     []string{"get"},
+					},
+				},
+			},
+		},
+	}
+
+	require.Equal(t, expectedDeniedPermissionsFooAwsapp, nestedApp.Status.Staged.ImagePermissionsDenied, "should have denied permissions for aws.acorn.io on foo.awsapp")
+
+	rmapp(ctx, app)
+
+	// --------------------------------------------------------------------
+	// Run #4 - Expect no denied permissions since we have an IRA that covers the image
+	// --------------------------------------------------------------------
+
+	// Add the missing api group to the IRA
+	ira.ImageSelector.NamePatterns = []string{tagName, nestedImageTagName}
```

Comment:
I thought the image selector was already set up for the parent image and the nested image. Why is it being modified here?

Reply:
Because I didn't check what I set it up for before :)
Changed.